sdk/scheduler/src/main/java/com/mesosphere/sdk/api/EndpointsResource.java

```diff
@@ -211,9 +215,27 @@ private static void addPortToEndpoints(
             // append entry to 'dns' and 'address' arrays for this task:
             vipEndpoint.append(RESPONSE_KEY_DNS, dnsHostPort);
             vipEndpoint.append(RESPONSE_KEY_ADDRESS, addressHostPort);
-            // populate 'vip' field if not yet populated (due to another task with the same vip):
-            vipEndpoint.put(RESPONSE_KEY_VIP, String.format("%s.%s.%s:%d",
-                    vipInfo.name, serviceName, ResourceUtils.VIP_HOST_TLD, vipInfo.port));
+
+            // (distinctly) append entry to 'vips' for this task:
+            String vipHostPort = String.format("%s.%s.%s:%d",
+                    vipInfo.name, serviceName, ResourceUtils.VIP_HOST_TLD, vipInfo.port);
+            Boolean foundVip = false;
+            if (vipEndpoint.has(RESPONSE_KEY_VIPS)) {
+                JSONArray vips = vipEndpoint.getJSONArray(RESPONSE_KEY_VIPS);
+                for (Object vipPresent : vips) {
```

Comment:
Nitpick: Could alternately do this and forego the typecast. Looks like JSONArray would have some additional internal checks for string types as well:
```
for (int i = 0; i < vips.length(); ++i) {
  if (vipHostPort.equals(vips.getString(i)) {
    foundVip = true;
    break;
  }
}
```